flow/project.py

```diff
@@ -1601,13 +1870,31 @@ def _generate_operations(self, cmd, jobs, requires=None):
             cmd_ = cmd.format(job=job)
             yield JobOperation(name=cmd_.replace(' ', '-'), cmd=cmd_, job=job)
 
-    def _get_pending_operations(self, jobs, operation_names=None):
+    def _get_pending_groups(self, jobs, groups, exec_mode=False):
+        if exec_mode:
+            mode = 'exec'
+        else:
+            mode = 'run'
```

Comment:
Try to convert to the str `mode` argument as soon as possible instead of passing on the boolean throughout the code.

Consider to use this idiom wherever you need to convert `mode='exec' if exec_mode else 'run'` for increased compactness.